### ISSUE TITLE
Stack::is_a

### ISSUE THREAD

**Issue by progschj** [vinniefalco/LuaBridge#44]
I think it would be useful to have a is_a function in the stack class, that allows to check whether the object at the given position has a specific type (mimicking the lua_isnumber etc. functions). I tried figuring out how to implement that (I assume it would essentially be similar to the get function without actually pulling the value?) but don't understand the details well enough.

**Comment by rysson** [comment]
Yes, yes, yes!

I need it. Thank you progschj.

Vinniefalo, could you merge it?

**Comment by vinniefalco** [comment]
I am not comfortable merging anything without testing, and I am not set up for LuaBridge development currently. Sorry about that!

**Comment by rysson** [comment]
I will be waiting :-)

BTW. Thanks for your work!


There is a small compile issue, patch:
```
--- a/Source/LuaBridge/detail/Stack.h
+++ b/Source/LuaBridge/detail/Stack.h
@@ -582,7 +582,7 @@ struct Stack <std::string const&>
 {
   static inline void push (lua_State* L, std::string const& str)
   {
-    lua_pushstring (L, str.c_str(), str.size());
+    lua_pushlstring (L, str.c_str(), str.size());
   }
 
   static inline std::string get (lua_State* L, int index)
```

**Comment by dmitry-t** [comment]
Implemented in #218.
